wp-includes/sqlite/class-wp-sqlite-translator.php

```diff
@@ -1550,14 +1550,41 @@ private function execute_describe() {
 	 */
 	private function execute_update() {
 		$this->rewriter->consume(); // Update.
-
-		$params = array();
+		$limit    = $this->rewriter->peek(
+			array(
+				'type'  => WP_SQLite_Token::TYPE_KEYWORD,
+				'value' => 'LIMIT',
+			)
+		);
+		$order_by = $this->rewriter->peek(
+			array(
+				'type'  => WP_SQLite_Token::TYPE_KEYWORD,
+				'value' => 'ORDER BY',
+			)
+		);
+		$where    = $this->rewriter->peek(
+			array(
+				'type'  => WP_SQLite_Token::TYPE_KEYWORD,
+				'value' => 'WHERE',
+			)
+		);
+		$params   = array();
 		while ( true ) {
 			$token = $this->rewriter->peek();
 			if ( ! $token ) {
 				break;
 			}
 
+			if ( $token->value === 'WHERE' && ( $limit || $order_by ) ) {
+				$this->remember_last_reserved_keyword( $token );
+				$this->rewriter->consume();
+				$this->prepare_update_for_limit_or_order();
+			}
+
+			if ( $token->value === ';' && $token->type === WP_SQLite_Token::TYPE_DELIMITER && ( $limit || $order_by ) ) {
```

Comment:
Why is this condition needed? Query processing should never advance past the `;` delimiter

Reply:
I didn't add it initially, but while I was writing the tests made me to add this statement.
What was happening is the following:
1. We have the query `SELECT * FROM test limit 10;`
2. While we are in the loop we add the where, so the above becomes: `SELECT * FROM test WHERE rowid IN ( SELECT rowid LIMIT 10;
3. After the loop I add the closing parenthesis, so the above becomes:
4. SELECT * FROM test WHERE rowid IN ( SELECT rowid LIMIT 10;)

Which is wrong. This is why I skip the `;`.

Reply:
Aha, gotcha! What if the query is just `SELECT * FROM test limit 10` without the final `;`, though?

Reply:
Maybe `;` should not be consumed and just `break` from the loop, and then you would add the closing `)` after the loop?

Also remember to make sure `UPDATE` query without `WHERE` keep working.

Reply:
> Also remember to make sure UPDATE queries without WHERE keep working.
I added some extra tests to ensure that.
I also added some nested select statements, to test if it executes the query as expected.
Finally, I break when I encounter a `;`.